indigo/indigo/src/main/scala/indigo/platform/storage/Storage.scala

```diff
@@ -1,23 +1,75 @@
 package indigo.platform.storage
 
+import indigo.shared.events.StorageActionType
+import indigo.shared.events.StorageEventError
+import indigo.shared.events.StorageEventError.FeatureNotAvailable
+import indigo.shared.events.StorageEventError.InvalidPermissions
+import indigo.shared.events.StorageEventError.QuotaExceeded
+import indigo.shared.events.StorageEventError.Unspecified
 import org.scalajs.dom
 
+import scala.scalajs.js
+
 final class Storage {
+  @SuppressWarnings(Array("scalafix:DisableSyntax.null"))
+  def key(index: Int): Either[Option[String], StorageEventError] =
+    try
+      if dom.window.localStorage == null then Right(FeatureNotAvailable(index, StorageActionType.Find))
+      else Left(Option(dom.window.localStorage.key(index)))
+    catch {
+      case e: js.JavaScriptException =>
+        Right(errToEvent(e, index, StorageActionType.Find))
+    }
 
-  def key(index: Int): Option[String] =
-    Option(dom.window.localStorage.key(index))
+  @SuppressWarnings(Array("scalafix:DisableSyntax.null"))
+  def save(key: String, data: String): Either[Unit, StorageEventError] =
```

Comment:
Not the end of the world, just threw me for a moment. The strong convention is for error's to be on the left, i.e. `Either[StorageEventError, Unit]`. This is because Either is 'right bias' meaning if you do `myEither.map(x => x)`, `x` will be the thing on the right of the either, either your happy path value, rather than the error.